arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -0,0 +1,271 @@
+        
+        let parameters = AGSUtilityTraceParameters(traceType: .isolation, startingLocations: [startingLocationElement])
+        parameters.traceConfiguration = configuration
+        return parameters
+    }
+    
+    func loadUtilityNetwork() {
+        setStatus(message: "Loading utility network…")
+        // Load the utility network to be ready to run a trace against it.
+        utilityNetwork.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.setStatus(message: "Loading utility network failed.")
+                self.presentAlert(error: error)
+            } else {
+                let networkDefinition = self.utilityNetwork.definition
+                let domainNetwork = networkDefinition.domainNetwork(withDomainNetworkName: self.domainNetworkName)
+                let utilityTier = domainNetwork?.tier(withName: self.tierName)
+                self.filterBarrierCategories = networkDefinition.categories
+                // Get the trace configuration for the specified utility tier.
+                self.traceConfiguration = utilityTier?.traceConfiguration
+                // Create a trace filter.
+                self.traceConfiguration?.filter = AGSUtilityTraceFilter()
+                
+                // Get a default starting location.
+                let networkSource = networkDefinition.networkSource(withName: self.networkSourceName)
+                let assetGroup = networkSource?.assetGroup(withName: self.assetGroupName)
+                let assetType = assetGroup?.assetType(withName: self.assetTypeName)
+                if let type = assetType, let element = self.utilityNetwork.createElement(with: type, globalID: self.globalId) {
+                    self.startingLocationElement = element
+                    // Draw the starting location element on the map.
+                    self.drawStartingLocation()
+                    self.setStatus(message: "Utility network loaded.")
+                    self.setUIState()
+                } else {
+                    self.setStatus(message: "Creating starting location failed.")
+                }
+            }
+        }
+    }
+    
+    // MARK: UI and feedback
+    
+    func drawStartingLocation() {
+        // Get a list of features for the starting location element.
+        utilityNetwork.features(for: [startingLocationElement]) { [weak self] (features, error) in
+            guard let self = self else { return }
+            if let features = features {
+                if let feature = features.first {
+                    // Get the geometry of the first feature for the starting location as a point.
+                    guard let startingLocationGeometry = feature.geometry as? AGSPoint else {
+                        self.setStatus(message: "Drawing starting location feature failed.")
+                        return
+                    }
```

Comment:
I suppose this is fine. It is a little unusual to have a `guard` statement nested in an `if` statement and not at the top level.

Reply:
Change it to `if let ...` to make it look consistent.